test/framework/src/main/java/org/elasticsearch/indices/recovery/AsyncRecoveryTarget.java

```diff
@@ -55,8 +55,8 @@ public void finalizeRecovery(long globalCheckpoint, long trimAboveSeqNo, ActionL
     }
 
     @Override
-    public void handoffPrimaryContext(ReplicationTracker.PrimaryContext primaryContext) {
-        target.handoffPrimaryContext(primaryContext);
+    public void handoffPrimaryContext(ReplicationTracker.PrimaryContext primaryContext, ActionListener<Void> listener) {
+        target.handoffPrimaryContext(primaryContext, listener);
```

Comment:
Wondering why this doesn't use the `executor` like the other methods in this class.

Reply:
Couldn't do that before because we needed a blocking API here, but now that we don't have that any longer I'll make it blocking, no more reason not to :)